torch/_functorch/partitioners.py

```diff
@@ -993,15 +1008,21 @@ def ban_recomputation_if_allowed(node):
         if op_types.is_view(node):
             return False
         if node in dont_ban:
-            return False
+            # collectives are *always* banned from recompute, overriding `dont_ban`
+            # (in particular, the activation memory budget logic is not allowed to recompute collectives)
+            is_collective = (
+                isinstance(node.target, torch._ops.OpOverload)
+                and node.target.namespace == "_c10d_functional"
```

Comment:
Is this the only type of collective we support? I have been away from inductor for a while but recently I noticed it has some support for inplace collectives as well as out of place so I'm not sure how that connects to the ops that dynamo can put into a compiled graph.

Reply:
We might want to add a "is_collective" tag to operators and then query that instead

Reply:
I guess on one hand, it would be nice if we had some strict invariants around e.g. only functional collectives in a given namespace being allowed through AOTAutograd. @wconstab do you know who we can check in with on the latest r.e. types of collectives in inductor?

On the other hand, there are lots of other places where we use tags as our primary way of grouping together ops into a category in the compiler. So happy to switch to the tag approach

Reply:
I'm fine with a special case on c10d_functional for now, it's not clear to me what a user-defined collective would look like

Reply:
@yifuwang @yf225 do you know if we convert functional collectives into inplace ones inside inductor? I think we only allow functional ones in the AOT graph but i'm not 100% sure